src/lib/containers/access_requirement_list/AccessRequirementList.tsx

```diff
@@ -1,62 +1,137 @@
 import { AccessRequirement } from '../../utils/synapseTypes/AccessRequirement/AccessRequirement'
 import { getAllAccessRequirements } from '../../utils/SynapseClient'
-import { SynapseConstants } from '../../utils/'
-import Modal from 'react-bootstrap/Modal'
-import SelfSignAccessRequirement from './SelfSignAccessRequirement'
+import { SynapseConstants, SynapseClient } from '../../utils/'
+import * as ReactBootstrap from 'react-bootstrap'
+import SelfSignAccessRequirementComponent from './SelfSignAccessRequirement'
 import { library } from '@fortawesome/fontawesome-svg-core'
 import { faCircle } from '@fortawesome/free-solid-svg-icons'
-import TermsOfUseAccessRequirement from './TermsOfUseAccessRequirement'
+import TermsOfUseAccessRequirementComponent from './TermsOfUseAccessRequirement'
+import ManagedACTAccessRequirementComponent from './ManagedACTAccessRequirement'
+import ACTAccessRequirementComponent from './ACTAccessRequirement'
+import { UserProfile } from '../../utils/synapseTypes'
+import useGetEntityHeaders, {
+  useGetEntityHeadersProps,
+} from '../../utils/hooks/useGetEntityHeaders'
+import AccessApprovalCheckMark from './AccessApprovalCheckMark'
 
 library.add(faCircle)
 
-type Props = {
+export type AccessRequirementListProps = {
   entityId: string
   token: string | undefined
+  accessRequirementFromProps?: Array<AccessRequirement>
   onHide?: Function
 }
 
-enum SUPPORTED_ACCESS_REQUIREMENTS {
+export enum SUPPORTED_ACCESS_REQUIREMENTS {
   SelfSignAccessRequirement = 'org.sagebionetworks.repo.model.SelfSignAccessRequirement',
   TermsOfUseAccessRequirement = 'org.sagebionetworks.repo.model.TermsOfUseAccessRequirement',
+  ManagedACTAccessRequirement = 'org.sagebionetworks.repo.model.ManagedACTAccessRequirement',
+  ACTAccessRequirement = 'org.sagebionetworks.repo.model.ACTAccessRequirement',
+}
+
+export const checkUnSupportedRequirement = (
+  accessRequirements: Array<AccessRequirement>,
+): boolean => {
+  let hasUnSupported = false
+  for (let i = 0; i < accessRequirements.length; i++) {
+    if (
+      accessRequirements[i].concreteType !==
+        SUPPORTED_ACCESS_REQUIREMENTS.TermsOfUseAccessRequirement &&
+      accessRequirements[i].concreteType !==
+        SUPPORTED_ACCESS_REQUIREMENTS.SelfSignAccessRequirement
+    ) {
+      hasUnSupported = hasUnSupported || true
+    }
+  }
+  return hasUnSupported
 }
 
 export default function AccessRequirementList({
   entityId,
   token,
   onHide,
-}: Props) {
+  accessRequirementFromProps,
+}: AccessRequirementListProps) {
   const [accessRequirements, setAccessRequirements] = useState<
     Array<AccessRequirement>
-  >([])
+  >(accessRequirementFromProps ?? [])
+
   const [isLoading, setIsLoading] = useState<boolean>(false)
+  const [user, setUser] = useState<UserProfile>()
+
+  const entityHeaderProps: useGetEntityHeadersProps = {
+    references: [entityId],
+    token: token,
+  }
+
+  const entityInformation = useGetEntityHeaders(entityHeaderProps)
```

Comment:
with alina's new PR, this should use the useGetInfoFromIds hook instead (you'll have to pull upstream to get it), it should like this 
```
const entityHeaderProps: useGetEntityHeadersProps = {
    ids: [entityId],
    token,
    type: 'ENTITY_HEADER'
  }
 const entityInformation = useGetInfoFromIds<EntityHeader>(entityHeaderProps)
```